File: ios/iosApp/UI/Screen/Screen1View.swift
import SwiftUI
import Shared
import Trikot

struct Screen1View: View {
    @ObservedObject private var observableViewModel: ObservableViewModelAdapter<Screen1ViewModel>

    init(viewModel: Screen1ViewModel) {
        observableViewModel = viewModel.asObservable()
    }

    var viewModel: Screen1ViewModel {
        observableViewModel.viewModel
    }

    var body: some View {
        VStack(spacing: 32) {
            VMDText(viewModel.title)

            Spacer()
            
            VMDButton(viewModel.closeButton) {
                Text($0.text)
            }

            VMDButton(viewModel.modalButton) {
                Text($0.text)
            }

            VMDButton(viewModel.pushButton) {
                Text($0.text)
            }

            VMDButton(viewModel.popToRoot) {
                Text($0.text)
            }

            VMDButton(viewModel.popToScreen3Inclusive) {
                Text($0.text)
            }

            VMDButton(viewModel.popToScreen3Exclusive) {
                Text($0.text)
            }

            Spacer()
        }
        .frame(maxWidth: .infinity)
        .background(Color.black.opacity(0.1))
    }
}

#Preview {
    Screen1View(viewModel: factoryPreview().createScreen1())
}

File: androidApp/src/main/java/com/mirego/kmp/boilerplate/app/ui/screens/Screen1View.kt
package com.mirego.kmp.boilerplate.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.mirego.compose.utils.SpacerVertical
import com.mirego.kmp.boilerplate.viewmodel.screen.Screen1ViewModel
import com.mirego.trikot.viewmodels.declarative.compose.extensions.observeAsState
import com.mirego.trikot.viewmodels.declarative.compose.viewmodel.VMDButton
import com.mirego.trikot.viewmodels.declarative.compose.viewmodel.VMDText

@Composable
fun Screen1View(screen1ViewModel: Screen1ViewModel) {
    val viewModel by screen1ViewModel.observeAsState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.LightGray),
        verticalArrangement = Arrangement.spacedBy(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        VMDText(viewModel = viewModel.title)
        SpacerVertical(height = 50.dp)
        VMDButton(
            modifier = Modifier.height(48.dp),
            viewModel = viewModel.closeButton,
        ) {
            Text(text = it.text)
        }
        VMDButton(
            modifier = Modifier.height(48.dp),
            viewModel = viewModel.pushButton,
        ) {
            Text(text = it.text)
        }
        VMDButton(
            modifier = Modifier.height(48.dp),
            viewModel = viewModel.modalButton,
        ) {
            Text(text = it.text)
        }
        VMDButton(
            modifier = Modifier.height(48.dp),
            viewModel = viewModel.popToRoot,
        ) {
            Text(text = it.text)
        }
        VMDButton(
            modifier = Modifier.height(48.dp),
            viewModel = viewModel.popToScreen3Inclusive,
        ) {
            Text(text = it.text)
        }
        VMDButton(
            modifier = Modifier.height(48.dp),
            viewModel = viewModel.popToScreen3Exclusive,
        ) {
            Text(text = it.text)
        }
    }
}
